bucket/foobar2000.json

```
@@ -28,22 +29,20 @@
             "    New-Item \"$persist_dir\\playlists\" -ItemType Directory -Force | Out-Null",
             "    if (Test-Path \"$env:Appdata\\foobar2000\") {",
             "        Write-Host -F yellow \"Copying old '$env:Appdata\\foobar2000' to '$persist_dir'\"",
-            "        Copy-Item \"$env:Appdata\\foobar2000\\*\" \"$dir\" -Recurse -Force",
+            "        Copy-Item \"$env:Appdata\\foobar2000\\*\" \"$dir\" -Force -Recurse",
             "    }",
             "}",
             "else {",
-            "    $shortVersion = $version.Split('.')[0, 1] -Join '.'",
-            "    Copy-Item \"$persist_dir\\playlists\" \"$dir\\playlist-v$shortVersion\" -Force -Recurse",
+            "    Copy-Item \"$persist_dir\\playlists\\*\" \"$dir\" -Force -Recurse",
             "}",
             "if (Test-Path \"$persist_dir\\theme.fth\") {",
             "    Copy-Item \"$persist_dir\\theme.fth\" \"$dir\" -Force",
             "}"
         ]
     },
-    "post_install": "if (is_directory \"$dir\\theme.fth\") { Remove-Item \"$dir\\theme.fth\" -Force }",
     "uninstaller": {
         "script": [
-            "Copy-Item \"$dir\\playlists-v*\\*\" \"$persist_dir\\playlists\" -Force  -Recurse",
+            "Copy-Item \"$dir\\playlists-v*\" \"$persist_dir\\playlists\" -Force  -Recurse",
```

```suggestion
            "Copy-Item \"$dir\\playlists-v*\" \"$persist_dir\\playlists\" -Force -Recurse",
```

My fault, sorry.
When update from 1.3 to 1.4, the struct is 
`persist_dir > playlist > playlist-v1.3` and `dir > playlist-v1.3` .
Then what will foobar do? Will it rename playlist-v1.3 to playlist-v1.4?
If then, we will get `persist_dir > playlist > playlist-v1.3, playlist-v1.4` and `dir > playlist-v1.3, playlist-v1.4` in the next update.

---

After a bit of testing, how it works is if `playlists-v1.3` is detected in version 1.4, foobar converts them to the new format used in 1.4 automatically. 

So `playlists-v1.3` does need to be moved during upgrades, but isn't actually needed after foobar has converted them (the converted playlists will be needed in `playlists-v1.4`. By default foobar doesn't do anything with the unused earlier playlist folder - after converting, it just leaves it.

---

Thanks, I got it. If I just copy playlists-v1.4 to $dir when update from 1.3 as what I did, foobar can't load playlist automatically. But now you also persist the outdate playlist. So I suggest that we only copy the current playlist.

```
    "installer": {
        "script": [
            "Remove-Item \"$dir\\`$*\", \"$dir\\uninstall.exe\" -Force -Recurse",
            "New-Item \"$dir\\portable_mode_enabled\" | Out-Null",
            "if(!(Test-Path \"$persist_dir\")) {",
            "    if (Test-Path \"$env:Appdata\\foobar2000\") {",
            "        Write-Host -F yellow \"Copying old '$env:Appdata\\foobar2000' to '$persist_dir'\"",
            "        Copy-Item \"$env:Appdata\\foobar2000\\*\" \"$dir\" -Recurse",
            "    }",
            "}",
            "else {",
            "    Move-Item \"$persist_dir\\playlists-v*\" \"$dir\"",
            "}",
            "if (Test-Path \"$persist_dir\\theme.fth\") {",
            "    Copy-Item \"$persist_dir\\theme.fth\" \"$dir\"",
            "}"
        ]
    },
    "uninstaller": {
        "script": [
            "$shortVersion = $version.Split('.')[0, 1] -Join '.'",
            "if (Test-Path \"$dir\\playlists-v$shortVersion\") {",
            "    Copy-Item \"$dir\\playlists-v$shortVersion\" \"$persist_dir\" -Recurse",
            "}",
            "if (Test-Path \"$dir\\theme.fth\") {",
            "    Copy-Item \"$dir\\theme.fth\" \"$persist_dir\"",
            "}"
        ]
    },
```

---

Sure - made the change.